Node.js/GettingStarted/processcsv.py

```diff
@@ -5,96 +5,77 @@
-                    
+                    # Creating a JSON object for every row
+                    line = "{"                    
                     for r in rows:
-                        if i == 0:
-                            
+                        if i == 0:                            
                             line = line + '"' + header[i] + '":"' + r.decode('utf-8','ignore') + '"'
-                        else:
-                            
+                        else:                            
                             line = line + "," + '"' + header[i] + '":"' + r.decode('utf-8','ignore') + '"'
                         i = i + 1
-
-                    line = line + "}"            
-                    
-
+                    line = line + "}" 
             all_vals.append(json.loads(json.dumps(line)))
             line = ""
         if not os.path.exists(outdir):
             os.makedirs(outdir)
-        json_fpath = outdir + "/" + outfname+'.json'
-        o = open(outdir + "/" + outfname+'.json',mode='w')
+        json_fpath = outdir + "/" + outfname + '.json'
+        o = open(json_fpath,mode='w')
         json.dump(all_vals,o)         
         o.close()
         return json_fpath
-        
-        
 
 if __name__ == "__main__":
-    ### TODO ### Replace day and month variable with calculating using datetime
 
     parser = argparse.ArgumentParser(description="Processes and stores data into hbase")
     parser.add_argument("--container",dest="container")
     parser.add_argument("--pattern",dest="pattern")    
-    args = parser.parse_args()
-
     
+    args = parser.parse_args()   
     container = args.container
-    if args.pattern is not None:
-        pattern = args.pattern
-    else:
-        pattern = None
+    pattern = args.pattern   
 
     print("Processing files from container : " + str(container))
     if pattern is not None:
-        generator = block_blob_service.list_blobs(container_name= container, prefix= pattern)
+        blob_list = block_blob_service.list_blobs(container_name=container, prefix=pattern)
     else:
-        generator = block_blob_service.list_blobs(container_name= container)
+        blob_list = block_blob_service.list_blobs(container_name=container)
     
-    for blob in generator:
+    for blob in blob_list:
         print("Processing blob : " + blob.name)
         blob_name = getfilename(blob.name)
         downloadedblob = "downloaded_" + blob_name 
         block_blob_service.get_blob_to_path(container_name=container,blob_name=blob.name, file_path=downloadedblob, open_mode='w')
         if pattern is not None:
-            json_outpath = processcsvfile(fname=downloadedblob,seperator="|",outfname=blob_name,outdir='jsonfiles/'+ container + "/" + pattern)
+            json_outpath = processcsvfile(fname=downloadedblob,seperator="|",outfname=blob_name,outdir='jsonfiles/' + container + "/" + pattern)
             print("uploading blob" + json_outpath)
```

Comment:
These missing spaces make me anxious about how this has been tested.  Shouldn't we have noticed things like "creating pool IDpool-2017..." and "uploading blobcon1output.json" when running the sample?